src/apiRegistry/index.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { ApiPromise } from '@polkadot/api';
+import { HttpProvider, WsProvider } from '@polkadot/rpc-provider';
+import { OverrideBundleType, RegistryTypes } from '@polkadot/types/types';
+
+import { Log } from '../logging/Log';
+import tempTypesBundle from '../override-types/typesBundle';
+import { SidecarConfig } from '../SidecarConfig';
+
+export class ApiPromiseRegistry {
+	// instances of ApiPromise for each defined URL
+	private static _instancesBySpecName: Map<string, ApiPromise> = new Map();
+	private static _instancesByUrl: Map<string, ApiPromise> = new Map();
+
+	/**
+	 * Get the ApiPromise instance for the given spec name.
+	 * @param specName The spec name to get the ApiPromise instance for.
+	 * @returns The ApiPromise instance for the given spec name.
+	 */
+
+	public static async initApi(url: string): Promise<ApiPromise> {
+		const { logger } = Log;
+		logger.info(`Initializing API for ${url}`);
+
+		if (!this._instancesByUrl.has(url)) {
```

Comment:
You can have multiple api instances to the same url - depends on what the nodes max rpc capacity is I believe!